scalanativelib/test/src/HelloNativeWorldTests.scala

```diff
@@ -49,38 +48,33 @@ object HelloNativeWorldTests extends TestSuite {
           Seq(Developer("lihaoyi", "Li Haoyi", "https://github.com/lihaoyi"))
       )
     }
-
-    object buildUTest extends Cross[BuildModuleUtest](matrix:_*)
-    class BuildModuleUtest(crossScalaVersion: String, sNativeVersion: String, mode: ReleaseMode)
+    object buildJUnit extends Cross[BuildModuleJUnit](matrix:_*)
+    class BuildModuleJUnit(crossScalaVersion: String, sNativeVersion: String, mode: ReleaseMode)
       extends BuildModule(crossScalaVersion, sNativeVersion, mode) {
       object test extends super.Tests {
-        override def sources = T.sources{ millSourcePath / 'src / 'utest }
-        def testFrameworks = Seq("utest.runner.Framework")
+        override def sources = T.sources{ millSourcePath / 'src / 'junit }
+        def testFrameworks = Seq("com.novocode.junit.JUnitFramework")
         override def ivyDeps = Agg(
-          ivy"com.lihaoyi::utest::0.7.5"
+          ivy"org.scala-native::junit-runtime::$sNativeVersion"  
         )
-      }
-    }
-
-    object buildScalaTest extends Cross[BuildModuleScalaTest](matrix:_*)
-    class BuildModuleScalaTest(crossScalaVersion: String, sNativeVersion: String, mode: ReleaseMode)
-      extends BuildModule(crossScalaVersion, sNativeVersion, mode) {
-      object test extends super.Tests {
-        override def sources = T.sources{ millSourcePath / 'src / 'scalatest }
-        def testFrameworks = Seq("org.scalatest.tools.Framework")
-        override def ivyDeps = Agg(
-          ivy"org.scalatest::scalatest-flatspec::3.2.0-M2",
-          ivy"org.scalatest::scalatest-shouldmatchers::3.2.0-M2"
+        override def compileIvyDeps = Agg(  
+          ivy"org.scala-native:::junit-plugin:$sNativeVersion"  
         )
       }
     }
+
     object buildNoTests extends Cross[BuildModuleNoTests](matrix:_*)
     class BuildModuleNoTests(crossScalaVersion: String, sNativeVersion: String, mode: ReleaseMode)
       extends BuildModule(crossScalaVersion, sNativeVersion, mode) {
       object test extends super.Tests {
-        override def sources = T.sources{ millSourcePath / 'src / "no-tests" }
-        def ivyDeps = Agg(ivy"com.lihaoyi::utest::0.7.5")
-        def testFrameworks = Seq("utest.runner.Framework")
+        override def sources = T.sources{ millSourcePath / "src" / "no-tests" }
+        def testFrameworks = Seq("com.novocode.junit.JUnitFramework")
+        override def ivyDeps = Agg(
+          ivy"org.scala-native::junit-runtime::$sNativeVersion"  
+        )
+        override def compileIvyDeps = Agg(  
+          ivy"org.scala-native:::junit-plugin:$sNativeVersion"  
+        )
```

Comment:
This looks like a repetitiion of the JUnit-variant. Maybe, we can have it in a shared trait?

Reply:
Extracted on a separate trait ✅